src/charm.py

```diff
@@ -119,6 +121,111 @@ def __init__(self, *args):
         )
         self._grafana_dashboards = GrafanaDashboardProvider(self)
 
+    def _install_gems(self) -> None:
+        """Trigger Discourse installation process."""
+        container = self.unit.get_container(SERVICE_NAME)
+        if not container.can_connect():
+            logger.info("Not ready to install")
+            return
+        if self.model.unit.is_leader():
+            self.model.unit.status = MaintenanceStatus("Installing discourse")
+            try:
+                commands = [
+                    (False, ["rm", "-rf", "/var/lib/gems/*"]),
+                    (False, ["gem", "install", "-n", "bin", "bundler"]),
```

Comment:
I think gem install requires network. In that case, should the HTTP proxy be set up for it?

Reply:
Good catch! I moved it back to the ROCK